src/main/java/org/sonarqube/gradle/SonarPropertyComputer.java

```diff
@@ -85,6 +87,22 @@ public class SonarPropertyComputer {
   private static final String TEST_SOURCE_SET_SUFFIX = "test";
   public static final String SONAR_PROJECT_BASE_DIR = "sonar.projectBaseDir";
 
+  /**
+   * Find test files given the path by looking for the keyword "test", for example:
+   * - script/test/run.sh
+   *          ^^^^
+   * But exclude not test related English words.
+   */
+  private static final Pattern TEST_FILE_PATH_PATTERN = Pattern.compile(
+    // Exclude valid English words ending with "test": contest, protest, detest, attest
+    "(?<!con|pro|de|at)" +
+      // Find path containing "test"
+      "test" +
+      // Exclude valid English words starting with "test":
+      // - testate, testator, testatrix, testament, testimonial, testimony, testiness, testy
+      "(?!ate|ator|atrix|ament|imonial|imony|iness|y)",
```

Comment:
Let's add the source or method used to generate this list of words so that we can make sure in the future that the list is up to date or needs to be challenged